src/dp_vnf.c

```diff
@@ -27,6 +43,64 @@ void dp_vnf_free(void)
 	dp_free_jhash_table(vnf_handle_tbl);
 }
 
+static int dp_add_vnf_value(const struct dp_vnf *vnf, const uint8_t ul_addr6[DP_VNF_IPV6_ADDR_SIZE])
+{
+	int ret;
+	uint8_t *vnf_ul_addr6;
+
+	vnf_ul_addr6 = rte_zmalloc("vnf_value_mapping", (size_t)DP_VNF_IPV6_ADDR_SIZE, RTE_CACHE_LINE_SIZE);
+	if (!vnf_ul_addr6) {
+		DPS_LOG_WARNING("VNF value allocation failed", DP_LOG_IPV6(ul_addr6));
+		return DP_ERROR;
+	}
+
+	memcpy(vnf_ul_addr6, ul_addr6, DP_IPV6_ADDR_SIZE);
+
+	ret = rte_hash_add_key_data(vnf_value_tbl, vnf, vnf_ul_addr6);
+	if (DP_FAILED(ret)) {
+		DPS_LOG_ERR("Cannot add VNF value and the corresponding underlying IPv6 address to table", DP_LOG_RET(ret));
+		return DP_ERROR;
```

Comment:
I am not sure how I missed this earlier, but you need to `rte_free(vnf_ul_addr6)` here

Reply:
Indeed, done